test/unit/sync-catalog-deployment.bats

```diff
@@ -848,6 +887,25 @@ load _helpers
   [ "${actual}" = 'http://RELEASE-NAME-consul-server:8500' ]
 }
 
+@test "syncCatalog/Deployment: consul service is used when client.enabled=false and global.tls.enabled=true and autoencrypt on" {
```

Comment:
The `CONSUL_HTTP_ADDR` and `CONSUL_CACERT` vars don't change based on auto encrypt so I don't think we need this test either.

The four permutations: clients enabled true/false x tls enabled true/false, are already tested by the four tests in this section.